lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java

```diff
@@ -83,11 +85,11 @@
       final LocalAddressProvider localAddressProvider,
       final TransportFactory tpFactory) {
       this(name, hostAddress, listeningPort, codec, errorHandler, orderingGuarantee, numberOfTries, retryTimeout,
-              localAddressProvider, tpFactory, RangeTcpPortProvider.Default);
+              localAddressProvider, tpFactory, getTcpProvider());
```

Comment:
In order to completely eliminate the `RangeTcpPortProvider.Default`, wouldn't we want to remove this constructor? That way, it has always to be injected.

@bgchun What's the long-term thinking on this?